dev/breeze/tests/test_docker_command_utils.py

```diff
@@ -0,0 +1,173 @@
+    mock_check_docker_permission.assert_called_with(True)
+    mock_run_command.assert_called_with(
+        ['docker', 'version', '--format', '{{.Client.Version}}'],
+        verbose=True,
+        suppress_console_print=True,
+        capture_output=True,
+        text=True,
+    )
+    mock_console.print.assert_called_with("Good version of Docker: 20.10.0.")
+
+
+@mock.patch('airflow_breeze.utils.docker_command_utils.check_docker_permission')
+@mock.patch('airflow_breeze.utils.docker_command_utils.run_command')
+@mock.patch('airflow_breeze.utils.docker_command_utils.console')
+def test_check_docker_version_higher(mock_console, mock_run_command, mock_check_docker_permission):
+    mock_check_docker_permission.return_value = False
+    mock_run_command.return_value.returncode = 0
+    mock_run_command.return_value.stdout = "21.10.0"
+    check_docker_version(verbose=True)
+    mock_check_docker_permission.assert_called_with(True)
+    mock_run_command.assert_called_with(
+        ['docker', 'version', '--format', '{{.Client.Version}}'],
+        verbose=True,
+        suppress_console_print=True,
+        capture_output=True,
+        text=True,
+    )
+    mock_console.print.assert_called_with("Good version of Docker: 21.10.0.")
+
+
+@mock.patch('airflow_breeze.utils.docker_command_utils.run_command')
+@mock.patch('airflow_breeze.utils.docker_command_utils.console')
+def test_check_docker_compose_version_unknown(mock_console, mock_run_command):
+    check_docker_compose_version(verbose=True)
+    expected_run_command_calls = [
+        call(
+            ["docker-compose", "--version"],
+            verbose=True,
+            suppress_console_print=True,
+            capture_output=True,
+            text=True,
+        ),
+    ]
+    mock_run_command.assert_has_calls(expected_run_command_calls)
+    mock_console.print.assert_called_with(
+        'Unknown docker-compose version. At least 1.29 is needed! \
+        If Breeze fails upgrade to latest available docker-compose version'
+    )
+
+
+@mock.patch('airflow_breeze.utils.docker_command_utils.run_command')
+@mock.patch('airflow_breeze.utils.docker_command_utils.console')
+def test_check_docker_compose_version_low(mock_console, mock_run_command):
+    mock_run_command.return_value.returncode = 0
+    mock_run_command.return_value.stdout = "1.28.5"
+    check_docker_compose_version(verbose=True)
+    mock_run_command.assert_called_with(
+        ["docker-compose", "--version"],
+        verbose=True,
+        suppress_console_print=True,
+        capture_output=True,
+        text=True,
+    )
+    
+    # mock_console.print.assert_called_with('You have too old version of docker-compose: 1.28.5!                 At least 1.29 is needed! Please upgrade!')
+    mock_console.print.assert_called_with('See https://docs.docker.com/compose/install/ for instructions.                 Make sure docker-compose you install is first on the PATH variable of yours.')
+
+
```

Comment:
The below function `test_check_docker_compose_version_ok ` is the one that is failing @potiuk 
I am not sure about where i am going wrong

Reply:
You can always do step-by-step-debugging in such case (this is where unit tests integrated with IDE are useful)

In this case the problem is that the regexp pattern does not match 2-segment version. It expects 3-segment one (1.29.0).

Reply:
So `version_extracted`was None and it actualy never printed anything.

Reply:
So the tests actually detected a potential error case - if 1.29 was passed - it neither printed "good" nor "unknown" (which I would expect actually in this case).

Reply:
oh, I see. I will check about integrating unit tests with IDE

Reply:
That's the best part !